File: Greedy/Python/Increasing Triplet Subsequence.py
# Problem link: https://leetcode.com/problems/increasing-triplet-subsequence/description
# Time: O(n)
# Space: O(1)
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        one = two = float('inf')

        for num in nums:
            if num <= one:
                one = num
            elif num <= two:
                two = num
            else:
                return True
        
        return False

File: Greedy/Java/Increasing Triplet Subsequence.java
// Problem link: https://leetcode.com/problems/increasing-triplet-subsequence/description
// Time: O(n)
// Space: O(1)
class Solution {
    public boolean increasingTriplet(int[] nums) {
        int one = Integer.MAX_VALUE, two = Integer.MAX_VALUE;

        for(int num : nums) {
            if(num <= one) {
                one = num;
            }
            else if(num <= two) {
                two = num;
            }
            else {
                return true;
            }
        }

        return false;
    }
}
